rust-runtime/aws-smithy-http-server/examples/pokemon-service/tests/helpers.rs

```diff
@@ -33,15 +80,53 @@ pub fn client() -> Client<
     aws_smithy_client::erase::DynConnector,
     aws_smithy_client::erase::DynMiddleware<aws_smithy_client::erase::DynConnector>,
 > {
+    let base_url = PokemonServiceVariant::Http.base_url();
     let raw_client = Builder::new()
         .rustls()
-        .middleware_fn(|mut req| {
-            let http_req = req.http_mut();
-            let uri = format!("http://localhost:13734{}", http_req.uri().path());
-            *http_req.uri_mut() = uri.parse().unwrap();
-            req
-        })
+        .middleware_fn(rewrite_base_url(base_url))
+        .build_dyn();
+    let config = Config::builder().build();
+    Client::with_config(raw_client, config)
+}
+
+// Returns a client that only talks through https and http2 connections.
+// It is useful in testing whether our server can talk to http2.
+#[allow(dead_code)]
+pub fn client_http2_only() -> Client<
+    aws_smithy_client::erase::DynConnector,
+    aws_smithy_client::erase::DynMiddleware<aws_smithy_client::erase::DynConnector>,
+> {
+    // Create custom cert store and add our test certificate to prevent unknown cert issues.
+    let mut reader = BufReader::new(File::open(TEST_CERT).expect("could not open certificate"));
+    let certs = rustls_pemfile::certs(&mut reader).expect("could not parse certificate");
+    let mut roots = tokio_rustls::rustls::RootCertStore::empty();
+    roots.add_parsable_certificates(&certs);
+
+    let connector = hyper_rustls::HttpsConnectorBuilder::new()
+        .with_tls_config(
+            tokio_rustls::rustls::ClientConfig::builder()
+                .with_safe_defaults()
+                .with_root_certificates(roots)
+                .with_no_client_auth(),
+        )
+        .https_only()
+        .enable_http2()
+        .build();
+
+    let base_url = PokemonServiceVariant::Https.base_url();
+    let raw_client = Builder::new()
+        .connector(DynConnector::new(Adapter::builder().build(connector)))
+        .middleware_fn(rewrite_base_url(base_url))
         .build_dyn();
     let config = Config::builder().build();
     Client::with_config(raw_client, config)
 }
+
+fn rewrite_base_url(base_url: String) -> impl Fn(Request) -> Request + Clone {
+    move |mut req| {
+        let http_req = req.http_mut();
+        let uri = format!("{base_url}{}", http_req.uri().path());
+        *http_req.uri_mut() = uri.parse().unwrap();
+        req
+    }
+}
```

Comment:
```suggestion
fn rewrite_base_url(base_url: String) -> Request {
    let http_req = req.http_mut();
    let uri = format!("{base_url}{}", http_req.uri().path());
    *http_req.uri_mut() = uri.parse().unwrap();
    req
}
```

This should work too.